HeterogeneousCore/ROCmServices/interface/numberOfDevices.h

```diff
@@ -0,0 +1,14 @@
+#ifndef HeterogeneousCore_ROCmServices_interface_numberOfDevices_h
+#define HeterogeneousCore_ROCmServices_interface_numberOfDevices_h
+
+namespace cms::rocm {
+
+  // Returns the number of ROCm devices
+  // The difference wrt. the standard ROCm function or
+  // cms::rocm::deviceCount() is that if ROCmService is disabled,
+  // this function returns 0.
+  int numberOfDevices();
```

Comment:
OK, no problem in removing it.

On the other hand, why don't we make the more common `cms::cuda::deviceCount()` return `0` instead of throwing if CUDA is not available or initialised ?

Reply:
The original point of this function was to return `0` in case the `CUDAService` is disabled in configuration (for cases where the use of the Service handle would be inconvenient). For any new code I'd rely on code paths making use of `process.options.accelerators` (i.e. SwitchProducer and module type resolver) for that functionality.

I'm fine with making `cms::cuda::deviceCount()` to return `0` instead of an exception for no-device case. The Alpaka code `cms::alpakatools::devices<Platform>().size()` should behave similarly, right?

Reply:
Yes, IIRC Alpalka and our tools should return `0` if there are no CUDA (or ROCm) GPUs.